src/controllers/message.js

```diff
@@ -78,42 +68,89 @@ module.exports = [
           };
 
           const message = await msgRepository.get(opt);
+          if (!message) {
+            const err = new Error(errors.MESSAGE_NOT_EXIST);
+            err.name = errors.MESSAGE_NOT_EXIST;
+            throw err;
+          }
 
           res.json(message);
         } catch (error) {
+          console.log(error);
           next(error);
         }
       },
 
       editMessage: async (req, res, next) => {
-        const query = {
-          _id: req.body.ID,
-        };
+        try {
+          const query = {
+            _id: req.body.ID,
+          };
 
-        const update = {
-          text: req.body.text,
-        };
+          const time = new Date(req.body.time);
 
-        const opt = {
-          query,
-          update,
-        };
+          const update = {
+            text: req.body.text,
+            time,
+          };
 
-        const ret = await msgRepository.update(opt);
-        res.json(ret);
+          const opt = {
+            query,
+            update,
+          };
+
+          const message = await msgRepository.get({
+            query,
+          });
+
+          if (!message) {
+            const err = new Error(errors.MESSAGE_NOT_EXIST);
+            err.name = errors.MESSAGE_NOT_EXIST;
+            throw err;
+          }
+
+          if (!(message.author === req.auth.id)) {
+            const err = new Error(errors.MESSAGE_AUTHOR_ID_MISMATCH);
+            err.name = errors.MESSAGE_AUTHOR_ID_MISMATCH;
+            throw err;
+          }
+
+          const ret = await msgRepository.update(opt);
+          res.json(ret);
+        } catch (error) {
+          next(error);
+        }
       },
 
       deleteMessage: async (req, res, next) => {
-        const query = {
-          _id: req.body.ID,
-        };
+        try {
+          const query = {
+            _id: req.body.ID,
+          };
 
-        const opt = {
-          query,
-        };
+          const opt = {
+            query,
+          };
+
+          const message = await msgRepository.get(opt);
 
-        const ret = await msgRepository.delete(opt);
-        res.json(ret);
+          if (!message) {
+            const err = new Error(errors.MESSAGE_NOT_EXIST);
+            err.name = errors.MESSAGE_NOT_EXIST;
+            throw err;
+          }
+
+          if (!(message.author === req.auth.id)) {
+            const err = new Error(errors.MESSAGE_AUTHOR_ID_MISMATCH);
+            err.name = errors.MESSAGE_AUTHOR_ID_MISMATCH;
+            throw err;
+          }
+
+          const ret = await msgRepository.delete(opt);
```

Comment:
I don't think there is much point to return the message you just deleted.

Reply:
true

Reply:
Okay I see what you are trying to do with that. We can leave it in for now then.